### ISSUE TITLE
How to plot loss curves in Tensorboard

### ISSUE THREAD

**Issue by chhigansharma** [matterport/Mask_RCNN#2714]
Can someone guide how to use tensorboard to look at learning curves, I really tried few things available but no graphs coming up.

**Comment by chhigansharma** [comment]
I have tried following commands in terminal while training model but got same error Every time = **"No dashboards are active for the current data set."**
tensorboard --logdir=Mask_RCNN\logs\boundary20211026T1821
tensorboard --logdir=Mask_RCNN\logs\boundary20211026T1821 --host=localhost
tensorboard --logdir=Mask_RCNN\logs\boundary20211026T1821 --host=127.0.0.1

Do we need to modify in mask rcnn code anywhere to plot loss curves ?

**Comment by nataliameira** [comment]
Hello @chhigansharma !

I used the tensorboard on my jupyter notebook.

For this you need to be in the directory containing the logs directory.

In my case the root directory. So I used the following code:

```
import datetime
%load_ext tensorboard
%tensorboard –logdir logs/ --port=8008
```

**Comment by chhigansharma** [comment]
Hello @nataliameira !!

I have tried this and still getting the same issue.

start_train = time.time()
model.train(dataset_train, dataset_val, 
            learning_rate=config.LEARNING_RATE, 
            epochs=20,
            layers='heads')
end_train = time.time()
minutes = round((end_train - start_train) / 60, 2)
print(f'Training took {minutes} minutes')

This is my training code, Have you used something like this as well ?


log_dir = "logs/" + datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
tensorboard_callback = tf.keras.callbacks.TensorBoard(log_dir=log_dir, histogram_freq=1)

and then adding 
custom_callbacks=[tensorboard_callback])

Thanks

**Comment by chhigansharma** [comment]
I'm training using 

TensorFlow 2.3.0; Keras 2.4.0

**Comment by nataliameira** [comment]
Hello @chhigansharma,

I trained using:

```
pip uninstall keras-nightly
pip uninstall -y tensorflow
pip install h5py==2.10.0
pip install tensorflow==1.15
pip install tensorflow-gpu==1.15.0
pip install keras==2.1.6
```

`python3 custom.py train --dataset=dataset --weights=coco`

I customized the `balloon.py` template for my multiclass problem. I trained with the versions I mentioned above. And then, on my **jupyter notebook**, in the root directory, I ran the code I gave you above.

It accesses the `logs `directory and provides the **tensorboard** graphics on the **jupyter notebook**.

**Comment by nataliameira** [comment]
The tensorboard consumes a lot of memory. You can check that it is not running in parallel on your system. If this happens, you will have problems.

If you use **Linux**, you can try:
`ps aux|grep tensorboard`

And if it's running in parallel, a PID number will appear.
So you can run:
`sudo kill -9 PID`

**Comment by chhigansharma** [comment]
Hello @nataliameira 

Thanks for suggestions.

I have modified training code for tensor flow 2.x compatibility
**I'm training on Windows**
[Link](https://github.com/ipython/ipykernel/issues/395#issuecomment-479787997)

I think it have something to do with the above **link.**

**Comment by nataliameira** [comment]
Ok @chhigansharma,
Good Luck!

**Comment by MayPhu** [comment]
My  Tensorboard is showing only the model graphs, but not the loss curves. How should I do it?
